src/pages/namespace/Explorer/Tree/NewWorkflow/index.tsx

```diff
@@ -65,13 +65,17 @@ const NewWorkflow = ({
   } = useForm<FormInput>({
     resolver: zodResolver(
       z.object({
-        name: fileNameSchema.and(
-          z
-            .string()
-            .refine((name) => !(unallowedNames ?? []).some((n) => n === name), {
+        name: fileNameSchema
+          .transform((enteredName) => addYamlFileExtension(enteredName))
+          .refine(
+            (enteredName) =>
+              !(unallowedNames ?? []).some(
+                (unallowedName) => unallowedName === enteredName
+              ),
```

Comment:
```suggestion
            (nameWithExtension) =>
              !(unallowedNames ?? []).some(
                (unallowedName) => unallowedName === nameWithExtension
              ),
```
For readability, I would make a distinction between enteredName and nameWithExtension. Easy to mix them up otherwise.